### ISSUE TITLE
Browersync/ web pack not watching for changes

### ISSUE THREAD

**Issue by karneaud** [tadejstanic/wp-hrm-webpack#2]
Hello,

I'm using your start kit to run a development project on a docker container with the exception that I run my source code and php files inside a subfolder ('src') I have the following. 

node: 8
yarn 0.24.6
nam 5

my config file

```
const path = require('path');

module.exports = {
  THEME_NAME: 'gblsf-parallax-one',
  PROXY_TARGET: '192.168.99.100:8000',
  HOST: 'localhost',
  PORT: 3000,
  PATHS: {
    src: unipath(path.resolve(__dirname, '../src')),
    compiled: unipath(path.resolve(__dirname, '../build')),
    modules: unipath(path.resolve(__dirname, '../node_modules')),
    base: unipath(path.resolve(__dirname, '../')),
  }
};

function unipath(base) {
  return function join() {
    const _paths = [base].concat(Array.from(arguments));
    return path.resolve(path.join.apply(null, _paths));
  }
}

```

my dev file

```
global.watch = true;

const { PATHS, PROXY_TARGET } = require('./env.config');
const utils = require('./utils.config');
const path = require('path');
const fsp = require('fs-promise');
const browserSync = require('browser-sync').create();
const webpack = require('webpack');
const webpackDevMiddleware = require('webpack-dev-middleware');
const webpackHotMiddleware = require('webpack-hot-middleware');
const htmlInjector = require('bs-html-injector');
const webpackConfig = require('../webpack.config');

const bundler = webpack(webpackConfig);

const bsOptions = {
  files: [{
    // js changes are managed by webpack
    match: [ PATHS.src() + '/theme-files/**/*.php'],
    // manually sync everything else
    fn: synchronize,
  }, ],

  proxy: {
    // proxy local WP install
    target: PROXY_TARGET,
    middleware: [
      // converts browsersync into a webpack-dev-server
      webpackDevMiddleware(bundler, {
        publicPath: webpackConfig.output.publicPath,
        noInfo: false,
      }),

      // hot update js && css
      webpackHotMiddleware(bundler),
    ],
  },

  //this gets annoying
  open: false,
};

browserSync.use(htmlInjector, { restrictions: ['#page'] });

function finalize(){
  utils.finalize();
}

function synchronize(event, file) {
  // activate html injector
  if (file.endsWith('php')) { htmlInjector(); }
  finalize();
}

(async() => {
  await fsp.emptyDir(PATHS.compiled());
  await utils.addMainCss();
  await utils.finalize();

  browserSync.init(bsOptions);
})()
```

on install I get

>  warning fsevents@1.1.2: The platform "linux" is incompatible with this module.
> node_1   | info "fsevents@1.1.2" is an optional dependency and failed compatibility check. Excluding it from installation.

the productions process is successful but...

When running node --harmony ./webpack-configs/webpack.dev.js

i get

> $ node --harmony ./webpack-configs/webpack.dev.js 

> node_1           | [18:00:37] [write-file-webpack-plugin] options { exitOnErrors: true,
> node_1           |   force: false,
> node_1           |   log: true,
> node_1           |   test: null,
> node_1           |   useHashIndex: true }
> node_1           | Finalizing.
> node_1           | (node:80) DeprecationWarning: loaderUtils.parseQuery() received a non-string value which can be problematic, see https://github.com/webpack/loader-utils/issues/56
> node_1           | parseQuery() will be replaced with getOptions() in the next major version of loader-utils.
> node_1           | [BS] [HTML Injector] Running...
> node_1           | [BS] Proxying: http://192.168.99.100:8000
> node_1           | [BS] Access URLs:
> node_1           |  -----------------------------------
> node_1           |        Local: http://localhost:3000
> node_1           |     External: http://172.19.0.2:3000
> node_1           |  -----------------------------------
> node_1           |           UI: http://localhost:3001
> node_1           |  UI External: http://172.19.0.2:3001
> node_1           |  -----------------------------------
> node_1           | [BS] Watching files...
> node_1           | [18:00:42] [write-file-webpack-plugin] compiler.outputFileSystem is "MemoryFileSystem".
> node_1           | [18:00:42] [write-file-webpack-plugin] compiler.options.devServer.outputPath is "/build/js".
> node_1           | [18:00:42] [write-file-webpack-plugin] stats.compilation.errors.length is "0".

I can access the server via 3000/ 3001 successfully 

However no files are being updated or watched apparently

Is it because of the fsevent error?

am I matching the right files in the browser sync options

**Comment by mydnicq** [comment]
Honestly, I have no idea why those errors are coming out. I made several projects with this package, also in docker container, without any problems.

Maybe there is a problem with fsevent, but this is just a warning. One thing that caught attention was your proxy setting 'PROXY_TARGET: '192.168.99.100:8000'. I think this should be set to localhost:8000, because everything is running in a container.

**Comment by karneaud** [comment]
@tadejstanic 

Did not know they were errors...the issue really was that the "compiling and reloading" were not being triggered when editing the files from the host. It triggered when I edited them in the container. I managed to get it working though by explicitly setting "watch" and "watch-poll" options for webpack and usePolling for browsersync. 

don't think it is working as expected but it seems to have done the trick.
